frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -0,0 +1,185 @@
+import {
+  Button,
+  Tooltip,
+  TextListItem,
+  TextListItemVariants,
+  TextContent,
+  Spinner,
+  Stack,
+  StackItem,
+  Split,
+  SplitItem,
+  Bullseye,
+} from '@patternfly/react-core';
+import OutlinedPlayCircleIcon from '@patternfly/react-icons/dist/esm/icons/outlined-play-circle-icon';
+import PauseIcon from '@patternfly/react-icons/dist/esm/icons/pause-icon';
+import PlayIcon from '@patternfly/react-icons/dist/esm/icons/play-icon';
+import DownloadIcon from '@patternfly/react-icons/dist/esm/icons/download-icon';
+import { PipelineRunTaskDetails } from '~/concepts/pipelines/content/types';
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import useFetchLogs from '~/concepts/k8s/pods/useFetchLogs';
+import usePodContainerState from '~/concepts/k8s/pods/usePodContainerState';
+import LogsTabStatus from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTabStatus';
+import { LOG_TAIL_LINES } from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/const';
+import DashboardCodeEditor from '~/concepts/dashboard/codeEditor/DashboardCodeEditor';
+import useCodeEditorAsLogs from '~/concepts/dashboard/codeEditor/useCodeEditorAsLogs';
+import { downloadFullPodLog } from '~/concepts/k8s/pods/utils';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+
+type LogsTabProps = {
+  task: PipelineRunTaskDetails;
+};
+
+const NO_POD_ERROR = new Error('No pod found, unable to show logs');
+
+const LogsTab: React.FC<LogsTabProps> = ({ task }) => {
+  const { namespace } = usePipelinesAPI();
+  const podName = task.runDetails?.status.podName ?? '';
+  const podNameError = !podName ? NO_POD_ERROR : undefined;
+  const { podLoaded, podError, podContainers, selectedContainer, setSelectedContainer } =
+    usePodContainerState(podName);
+  const [isPaused, setIsPaused] = React.useState(false);
+  const [logs, logsLoaded, logsError, refreshLogs] = useFetchLogs(
+    podName,
+    selectedContainer?.name ?? '',
+    !isPaused,
+    LOG_TAIL_LINES,
+  );
+  const [downloading, setDownloading] = React.useState(false);
+  const [downloadError, setDownloadError] = React.useState<Error | undefined>();
+  const { scrollToBottom, onMount, editorOptions } = useCodeEditorAsLogs();
+
+  const scrollEditorToBottom = React.useCallback<typeof scrollToBottom>(() => {
+    if (!isPaused) {
+      scrollToBottom();
+    }
+  }, [isPaused, scrollToBottom]);
+
+  React.useEffect(() => {
+    if (!isPaused && logs) {
+      scrollEditorToBottom();
+    }
+  }, [isPaused, logs, scrollEditorToBottom]);
+
+  const canDownload = !!selectedContainer && !!podName && !downloading;
+  const onDownload = () => {
+    if (!canDownload) {
+      return;
+    }
+    setDownloadError(undefined);
+    downloadFullPodLog(namespace, podName, selectedContainer.name)
+      .then(() => setDownloading(false))
+      .catch((e) => {
+        setDownloading(false);
+        setDownloadError(e);
+      });
```

Comment:
`setDownloading(true)` is never called.
```suggestion
    setDownloading(true);
    downloadFullPodLog(namespace, podName, selectedContainer.name)
      .catch((e) => {
        setDownloadError(e);
      })
      .finally(() => {
        setDownloading(false);
      });
```